logs_sdk/lib/opentelemetry/sdk/logs/read_write_log_record.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # A {LogRecord} interface that can write to the full {LogRecord} and
+      # retrieve all information added to the {LogRecord}.
+      class ReadWriteLogRecord < OpenTelemetry::Logs::LogRecord
+        attr_accessor :timestamp,
+                      :observed_timestamp,
+                      :trace_id,
+                      :span_id,
+                      :trace_flags,
+                      :severity_text,
+                      :severity_number,
+                      :body,
+                      :attributes,
+                      :resource,
+                      :information_scope
+
+        # Creates a new {ReadWriteLogRecord}.
+        #
+        # @param [optional Float, Time] timestamp Time when the event occurred.
+        # @param [optional Float, Time] observed_timestamp Time when the event was observed.
```

Comment:
Float - The `Process.clock_gettime` return value's class
Time - The documented param class for `timestamp` on a Span is the `Time` class

Is there a preference for the type of object passed here? My guess would be `Process.clock_gettime` as it seems that many gems replaced their calls to `Time.now` with `Process.clock_gettime` in 2021's 1.0 release.

Reply:
Whatever happens here should also be applied to ReadableLogRecord